### DISCUSSION TITLE
Full compiled .css

### DISCUSSION THREAD

**Discussion by dautb** [tailwindlabs/tailwindcss#9387] [Help]
Where for 3.1?

**Comment by wongjn** [comment]
If you are asking about generating a CSS file of all possible classes in Tailwind v3.1, then I'm afraid there isn't one. This is due to the fact that Tailwind uses a just-in-time compiler to create classes only when needed. Furthermore, there would be an infinite amount of classes, due to arbitrary values and arbitrary variants which would be impossible to generate all combinations of. Even without these, if you really wanted every single class possible, you would need to generate every possible combination of the variants, (like `dark:first:hover:focus:md:visited:last:only:…:m-1`) so there would be all the "base" classes plus all these permutations of variants and thus the total number of possible classes would be something like (54![^1] + 1) × number of "base" classes (maths probably not correct here, but something like this). The "real" number would be slightly less since some variants don't make sense together, i.e. `sm:md:lg:xl:m-1` but this obscenely large number should demonstrate the point.

[^1]: This is 54 factorial, equal to 230,843,697,339,241,380,472,092,742,683,027,581,083,278,564,571,807,941,132,288,000,000,000,00

**Reply by dautb** [reply]
Or how else would you use this with a custom PHP site?

**Reply by wongjn** [reply]
I would have Tailwind scan the PHP code that contain Tailwind classes, by configuring the [`content` in `tailwind.config.js`](https://tailwindcss.com/docs/content-configuration) to build the CSS, and then deploy the CSS as needed.

**Comment by dautb** [comment]
Seems I can just include the .js file no? haha. 

Kind of silly to make something like that.

**Comment by dautb** [comment]
Sounds like a malware style approach. I would prefer that tailwind is isolated by itself, and does not require involvement in a whole project, rather just as a include.  Maybe it could potentially become modular someday, or someone code something similar without the excess.
